scripts/india_wris/base.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pandas as pd
+import json
+import re
+from india.geo.districts import IndiaDistrictsMapper
+
+
+class WaterQualityBase():
+
+    def __init__(self, dataset_name, util_names, template_strings):
+        self.dataset_name = dataset_name
+        self.util_names = util_names
+        self.module_dir = os.path.dirname(__file__)
+
+        self.solute_mcf = template_strings['solute_mcf']
+        self.solute_tmcf = template_strings['solute_tmcf']
+        self.chemprop_mcf = template_strings['chemprop_mcf']
+        self.chemprop_tmcf = template_strings['chemprop_tmcf']
+        self.site_dcid = template_strings['site_dcid']
+        self.unit_node = template_strings['unit_node']
+
+    def create_dcids_in_csv(self):
+        """
+        Method to map the district names to LGD District Codes
+        Mapped codes are used to create dcids for water quality stations
+        
+        Format of dcid for a station:
+            'india_wris/<lgd_code_of_district>_<name_of_station>'
+        Example: 'india_wris/579_Velanganni' for Velanganni station in
+                    Nagercoil, Tamil Nadu, India
+        """
+
+        self.df = pd.read_csv(
+            os.path.join(self.module_dir,
+                         'data/{}.csv'.format(self.util_names)))
+        # Dropping rows with all data missing
+        self.df.dropna(thresh=11, inplace=True)
```

Comment:
How was the value 11 chosen? Can it be derived, rather than hard-coded?

Also, were the dropped values inspected, and what were the issues with those rows? For processing scripts like this, it's best to print some reports as it runs, such as number of dropped rows, etc.

Reply:
There were rows in the csv in which only the place names and latlongs are present. All columns corresponding water quality properties were empty. So I checked the files and chose 11 as common value for all datasets, which is the `minimum number of non-empty columns + 1`.

Now I added a [helper method](https://github.com/datacommonsorg/data/blob/9e1ab51d6e1551ff631a7f65492e92af9f7d8c26/scripts/india_wris/base.py#L60) to the class to calculate the minimum number of non-empty columns and drop rows with empty columns equal to that value.